### DISCUSSION TITLE
validation error

### DISCUSSION THREAD

**Discussion by lvsltz** [dandi/helpdesk#43] [General]
Hi,

I would like to update dataset [25](https://gui.dandiarchive.org/#/dandiset/000025) with a newer file (discussion [here](https://github.com/NeurodataWithoutBorders/pynwb/issues/1335#issuecomment-983870265)) but validation fails with the following error.

```
[ERROR   ] dandi 40024:4524006912  
Failed to validate ./001_140709EXP_A1.nwb: data type 'NDXICEphysFile' not found in namespace core
```
The file is indeed created with an extension but there's no issue whatsoever if I load it with `load_namespaces = True` in pynwb.
`python -m pynwb.validate this_file` also doesn't report any issues.

How can I bypass the validation performed by the uploader ?

**Comment by bendichter** [comment]
Have you tried upgrading dandi?

`pip install -U dandi`

**Comment by lvsltz** [comment]
Hm, I installed dandi for the first time today, but I have upgraded once more now. 
I see the same issue.

**Comment by bendichter** [comment]
@oruebel do you know what might be going on here?

**Reply by oruebel** [reply]
@lvsltz all types from the icephys extension have been merged with PyNWB. Could you create the file without the icephys extension? If you are using your own extension in addition, it should be fairly simple to update it since all that should need to change is that you will need to include the types you need directly from the NWB core instead of the icephys extension.

**Reply by lvsltz** [reply]
I am using only core types (pynwb 2, hdmf 3.1.1, nwb-schema 2.4). I do have another extension that overrides `NWBFile` class instead.

I think that I am seeing this issue because dandi might be using a different validator (i.e. not pynwb's): for example dandi was also asking for explicit `subject_id`, whereas pynwb.validate doesn't. 
Since this is an example NWB file, not a data release, I think it should be enough if it's working with pynwb...

**Reply by oruebel** [reply]
Thanks for the clarification. I misunderstood issue. Ok, ``NDXICEphysFile`` is a type from your extension and it looks like PyNWB simply can't find the extension. The fact that ``python -m pynwb.validate`` works but ``dandi`` validation fails seems to indeed indicate that the issue is with how dandi calls the validator. The error looks like it is probably raised from PyNWB, because it can't find the ``NDXICEphysFile`` type. My guess is that the ``load_namespaces = True`` option may not be set correctly when dandi tries to validate so the namespace of your extension is not loaded.

**Comment by lvsltz** [comment]
Hello, @yarikoptic @satra @bendichter can we have a chat during the hackathon to better understand this issue?

**Reply by oruebel** [reply]
Happy to chat. I sent you an email to try an find a time.
